### PR TITLE
[CDLS] add rate limit

### PR DESCRIPTION
**Pull request by klindziukp**
## PROBLEM
- Need to enable rate limit to prevent from extra calls
## SOLUTION
- Add bucket4j
## TODO
- Use Redis

### CODE REVIEW

File: src/main/java/com/codeless/api/automation/storage/RateLimitStorage.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.codeless.api.automation.storage;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class RateLimitStorage {
```
Comment (Strunevskiy) [inline src/main/java/com/codeless/api/automation/storage/RateLimitStorage.java:6]: if you want you inject redis client here and store that data in Redis. As of now, it is ok, but It is good to have redis here.

**Review by Strunevskiy — APPROVED** [review]
Please create followup PR to have limit per API and per Limit, we can discuss how to do it.

File: src/main/java/com/codeless/api/automation/configuration/interceptor/RateLimitRequestInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  private RateLimitData getRateLimitData(HttpServletRequest request, HttpServletResponse response,
+      String username) {
+    RateLimitData rateLimitData = RateLimitStorage.getRateLimitData(username);
+    if (Objects.isNull(rateLimitData)) {
+      final User userByEmail = userService.getUserByEmail(request.getUserPrincipal().getName());
+      if (Objects.isNull(userByEmail)) {
+        response.setStatus(HttpStatus.BAD_REQUEST.value());
+        return null;
+      }
+      final Bucket bucket = pricingPlanService.resolveBucketByUserPlan(userByEmail.getUserPlan());
```
[the remark this answers is not in the document]
Comment (klindziukp) [reply src/main/java/com/codeless/api/automation/configuration/interceptor/RateLimitRequestInterceptor.java:64]: Let's discuss it during call.

File: src/main/java/com/codeless/api/automation/storage/RateLimitStorage.java (hunk 1 above)
Comment (klindziukp) [reply src/main/java/com/codeless/api/automation/storage/RateLimitStorage.java:6]: Yep! As wee agreed - Redis will be follow up step.

**Pull request merged**